### DISCUSSION TITLE
Trigger Notification Action in UI Tests

### DISCUSSION THREAD

**Discussion by bauersimon** [microsoft/vscode-discussions#1076] [Extension Development QnA] [answered]
Our extension triggers [notifications](https://code.visualstudio.com/api/ux-guidelines/notifications) with actions (buttons) using the LSP [`window/showMessageRequest`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#window_showMessageRequest).

How can I simulate that a user clicks on an action button in some UI test? I have already searched through the [VSCode API](https://code.visualstudio.com/api/references/vscode-api) and stackoverflow without any success on how to do this natively.

I am aware that there are some [third-party frameworks](https://github.com/redhat-developer/vscode-extension-tester) with this capability, but I wonder if it's doable natively without introducing a whole new different testing setup.

Thanks in advance for any hints how to achieve this. :pray:

**Comment by isidorn** [comment]
Not easy to do. You could use some custom library to try to achieve this.

**Reply by bauersimon** [reply]
Thank you @isidorn very much for the response.

Pardon me, but "not easy to do" is not a very satisfying answer here. Is it doable natively, **yes**? And if so, how exactly could it be achieved? Or is it **not** doable at all out of the box?

I guess with "custom library", you are pointing me towards [libraries that inspect the VSCode UI](https://github.com/redhat-developer/vscode-extension-tester),  which is again not the native solution I am looking for and would imply that no native solution exists.

I would be very grateful for a definite clarification on this matter.

**Comment by abrilurena** [comment]
Hi, @bauersimon! i hope you're doing great. It's been a while, do you still need help with this?

**Comment by abrilurena** [comment]
Hi, @bauersimon! I hope you're doing greaaat ✨ It has been a while since we last heard from you. Therefore, I will be closing this. However, please feel free to reach out or reopen this discussion if you still need help 😄

**Reply by bauersimon** [reply]
Sorry for the late reply. Thanks for getting back to me. As I tried to explain above, I would've liked a definite yes/no answer - not "Not easy to do.". So it there is still some clarification you could provide, I would be very thankful. Can notification actions be triggered during testing natively?

**Reply by abrilurena** [reply]
Hi, @isidorn! could you please help with this? thank you ✨

**Reply by isidorn** [reply, marked as answer]
No. Not possible using VS Code API.
